frontend/src/components/dialog/SettingsForm.vue

```diff
@@ -0,0 +1,171 @@
+              <slot name="error" />
+            </v-alert>
+
+            <!-- error message via props -->
+            <server-error v-else :server-error="error" />
+          </v-card-text>
+
+          <slot name="afterErrors" />
+
+          <div class="d-flex flex-wrap">
+            <v-card-actions>
+              <slot name="moreActions" />
+            </v-card-actions>
+            <v-spacer />
+            <v-card-actions>
+              <v-spacer />
+              <v-btn
+                v-if="cancelVisible && cancelAction != null"
+                :color="cancelColor"
+                text
+                :disabled="!cancelEnabled"
+                @click="doCancel"
+              >
+                {{ cancelLabel }}
+              </v-btn>
+              <v-btn
+                v-if="submitAction != null"
+                :color="submitColor"
+                type="submit"
+                :loading="isSaving"
+                :disabled="!submitEnabled"
+              >
+                <v-icon v-if="!!submitIcon" left>
+                  {{ submitIcon }}
+                </v-icon>
+                {{ submitLabel }}
+              </v-btn>
+              <slot name="actions" />
+            </v-card-actions>
+          </div>
+        </v-card>
+      </v-form>
+    </ValidationObserver>
+  </v-bottom-sheet>
+</template>
+
+<script>
+import { ValidationObserver } from 'vee-validate'
+import ServerError from '@/components/form/ServerError.vue'
+
+export default {
+  name: 'SettingsForm',
+  components: { ValidationObserver, ServerError },
+  props: {
+    value: { type: Boolean, required: true },
+
+    icon: { type: String, default: '', required: false },
+    title: { type: String, default: '', required: false },
+
+    submitAction: { type: Function, default: null, require: false },
+    submitIcon: { type: String, default: 'mdi-send-variant', required: false },
+    submitLabel: {
+      type: String,
+      default: function () {
+        return this.$tc('global.button.submit')
+      },
+      required: false,
+    },
+    submitColor: { type: String, default: 'primary', required: false },
+    submitEnabled: { type: Boolean, default: true, required: false },
+
+    cancelAction: { type: Function, default: null, required: false },
+    cancelIcon: { type: String, default: 'mdi-window-close', required: false },
+    cancelLabel: {
+      type: String,
+      default: function () {
+        return this.$tc('global.button.cancel')
+      },
+      required: false,
+    },
+    cancelColor: { type: String, default: 'secondary', required: false },
+    cancelEnabled: { type: Boolean, default: true, required: false },
+    cancelVisible: { type: Boolean, default: true, required: false },
+
+    loading: { type: Boolean, default: false, required: false },
+    error: { type: [Object, String, Error], default: null, required: false },
+
+    maxWidth: { type: String, default: '600px', required: false },
```

Comment:
i have the feeling that this component does too much.

Reply:
This is the same as DialogForm. But I think this could be simplified. I did not want to remove anything before we agree on this.